File: pages/login_page.py
from pages.base_page import BasePage

class LoginPage(BasePage):
    USERNAME_INPUT = "#email"
    PASSWORD_INPUT = "#pass"
    LOGIN_BUTTON = "[name='login']"

    def enter_username(self, username):
        self.enter_text(self.USERNAME_INPUT, username)

    def enter_password(self, password):
        self.enter_text(self.PASSWORD_INPUT, password)

    def click_login(self):
        self.click(self.LOGIN_BUTTON)

    def login(self, username, password):
        self.enter_username(username)
        self.enter_password(password)
        self.click_login()

File: src/pages/loginPage.js
class LoginPage {
    constructor(page) {
        this.page = page;
        this.usernameInput = 'input[name="email"]'; // Update selector as needed
        this.passwordInput = 'input[name="pass"]'; // Update selector as needed
        this.loginButton = 'button[name="login"]'; // Update selector as needed
    }

    async login(username, password) {
        await this.page.fill(this.usernameInput, username);
        await this.page.fill(this.passwordInput, password);
        await this.page.click(this.loginButton);
    }
}

export default LoginPage;